### ISSUE TITLE
[1.16.5] Disabling Ore Generation doesn't actually disable it. (though only sometimes)

### ISSUE THREAD

**Issue by Gbergz** [Da-Technomancer/Crossroads#140]
**Describe the bug**
When disabling the generation of Copper or Tin it doesn't disable it. (Though only sometimes, which is weird.)

**To Reproduce [Optional]**
Steps to reproduce the behavior: (Not really sure how to reproduce as its not happening 100% of the time.
1. Disable Copper, Tin Ore generation in configs.
2. Make a world
3. Still able to find copper and tin.
4. Annoying.

**Expected behavior [Optional]**
That they're disabled like in the config.

**MC:** 1.16.5
**Forge:** 36.0.13
**Crossroads version:** 2.6.3

**-** Gbergz

**Comment by TheDeviantCrafter** [comment]
I can confirm. The issue doesn't appear to be seed dependent, as I was able to create two worlds with the same seed and only one generated Crossroads ores.

If it helps, I've uploaded two worlds. One has ores, one doesn't. They were made with the same pack and the same seed. Maybe there's some useful info hidden in the level.dat file or something.

[No ores.zip](https://github.com/Da-Technomancer/Crossroads/files/6058405/No.ores.zip)
[Yes ores.zip](https://github.com/Da-Technomancer/Crossroads/files/6058406/Yes.ores.zip)

**Comment by Da-Technomancer** [comment]
For the world which generated with ores even when they were disabled, did you load any other worlds before creating that one? For example, did you open a pre-existing world, play it for a bit, close it, then create the new world that shouldn't have had ores, all without restarting MC?

Trying to narrow down the possible cause.

**Comment by TheDeviantCrafter** [comment]
The bug triggers upon reloading a world without restarting a pack. After closing to the title screen and entering a world, every world will generate ores. Old and new worlds are both affected.

**Comment by Gbergz** [comment]
> The bug triggers upon reloading a world without restarting a pack. After closing to the title screen and entering a world, every world will generate ores. Old and new worlds are both affected.

Can confirm that is the case. Creating a world upon starting the pack. No Crossroads Ores- disabled like the config.
But when I quit from that world and create another one they generate.
![image](https://user-images.githubusercontent.com/26634554/110528287-47e01c80-8118-11eb-8535-635cdb770536.png)
![image](https://user-images.githubusercontent.com/26634554/110528333-53334800-8118-11eb-8fd4-ef52bb63579a.png)

**Comment by Gbergz** [comment]
Also found out that a .bak file gets generated each time a worlds generate. Might have something to about this issue.
![image](https://user-images.githubusercontent.com/26634554/111043771-a53fda80-8444-11eb-83b5-2d4a5def6e8d.png)

(In the serverconfig folder located inside the world folder)

Gets generated each time I enter the world.

**Comment by Da-Technomancer** [comment]
If anyone needs/wants a mod file with this bugfix before I've published it on CurseForge, message me on the Crossroads discord and I can send you a compiled file.

**Comment by Gbergz** [comment]
Any ETA on when it's uploaded to CurseForge?

**Comment by Da-Technomancer** [comment]
@Gbergz Now. https://www.curseforge.com/minecraft/mc-mods/crossroads-mc/files/3246384

**Comment by Gbergz** [comment]
Seems to not be fixed. Still happening.
![image](https://user-images.githubusercontent.com/26634554/112669930-91db3900-8e60-11eb-889d-df007906768e.png)

Generating one on each load.

**Comment by Da-Technomancer** [comment]
Is the ore generation fixed? Because that should be working now. It's possible it was two separate issues.

**Comment by Gbergz** [comment]
Not fixed. Config only works when creating a new world on first launch. If you quit from that world and create another. Disabled Ores generate again. (Config isnt respected) Like I originally reported.

**Comment by Gbergz** [comment]
Also there's a server config in normal config folder + a server config in the serverconfig folder?

**Comment by Da-Technomancer** [comment]
The update fixed the issue in my testing- I'm no longer able to reproduce the bug.
I'd suggest updating your Forge to the latest version, and let me know if it's still happening for you.

**Comment by Gbergz** [comment]
I am on the latest version. Forge **36.1.2** as am writing this.

Crossroads version: 2.6.4

Second world creation:
![image](https://user-images.githubusercontent.com/26634554/112764272-10260f80-9008-11eb-9db6-965ccdf27455.png)


Edit:
I do get this in the logs
`\config\crossroads-server.toml is not correct. Correcting `
But the only thing I've done is to disable Tin & Copper ore generation. works the first the then 2nd world it becomes "incorrect".

Edit 2:
Also, which config am I suppose to even edit. The one that gets generated when creating a world inside the "serverconfig" folder (crossroads-server.toml). Or the exact same one in the regular config folder (crossroads-server.toml) ?. Right now I've edited the files in both locations. (Carried over the one in `serverconfig` to `defaultconfigs` folder)

**Comment by Da-Technomancer** [comment]
> Also, which config am I suppose to even edit. The one that gets generated when creating a world inside the "serverconfig" folder (crossroads-server.toml). Or the exact same one in the regular config folder (crossroads-server.toml) ?. Right now I've edited the files in both locations. (Carried over the one in serverconfig to defaultconfigs folder)

It's the new Forge config system that keeps confusing people. Every world has its own set of config files (in the `serverconfig` folder).  When you make a new world, it copies the config in the `defaultconfigs` folder into the newly created world's serverconfig file. 
So, for existing worlds, modify the `serverconfig`, and for changing the config before creating a new world, modify the `defaultconfig` file. 

Is this unnecessarily confusing? Yes. Is it completely unexplained? Yes. It's the new Forge config system that allows every world to have its own set of configs.

**Comment by Gbergz** [comment]
Yeah thanks. I know and understand the new config system. But there's two of the exact same file. The one in /config/ like old times. And another one in /serverconfig/. (Both are crossroads-server.toml) Which is a first. Usually it's client config in /config/ & server side configs in serverconfig. Or there's nothing in the serverconfig folder and everything is in /config/ like old times. Hope you understand.

**Comment by Da-Technomancer** [comment]
Totally unable to reproduce this issue at this point.

All I can suggest is that the `\config\crossroads-server.toml is not correct. Correcting` message, and large number of .bak files, are generally associated with incorrect config files. It's possible the config files themselves are malformed; deleting every crossroads config file (/serverconfig/, world specific, general, /config/, .toml, and .toml.bak; all of them) _might_ fix it for you, but that's more of a guess than a guarantee.

**Comment by Gbergz** [comment]
They're completely newly generated with only Tin Ore & Copper Ore disabled. I'm at a loss aswell and don't know why it's happening. Oh well! Thanks for trying to help.. appreciated.

Edit:
Here's a zip file containing the file that gets generated each time. Might help.
[crossroads-server-1.toml.zip](https://github.com/Da-Technomancer/Crossroads/files/6224013/crossroads-server-1.toml.zip)

Edit 2:
Also worth mentioning its happening on our official servers aswell.
![image](https://user-images.githubusercontent.com/26634554/112888940-467d9080-90d5-11eb-89ee-5b1cfe26c80c.png)

**Comment by TheDeviantCrafter** [comment]
Disabled ores are now generating even on fresh worlds right after a pack restart. I can't find any situation where the config setting applies correctly.

**Comment by Da-Technomancer** [comment]
I am aware that _multiple_ times in the past, I've said 'it's fixed, for real this time', and each time I was wrong.

However, hubris notwithstanding, it's fixed for real this time. 
This issue should be fixed with Crossroads 2.6.5. 
If updating doesn't fix the error, please re-open this issue.

**Comment by techwulf** [comment]
I'm still having the same issue...
Crossroads 2.6.5
Forge 36.2.33
I disable tin and copper gen in the main crossroads-server.toml file in the config folder, but they still generate in the world.
I've started multiple new worlds to test.
In each of the worlds serverconfig/crossroads-server.toml files it shows the ores as enabled, not disabled, despite having it disabled in the main settings file.

I'm editing because I've noticed all *-server.toml files aren't being respected in the config folder for any mod. It must be a Forge issue and not a Crossroads issue then.
